lib/src/common/disposable.dart

```diff
@@ -281,21 +281,54 @@ class Disposable implements _Disposable, DisposableManagerV6, LeakFlagger {
     return size;
   }
 
+  /// Whether the disposal of this object is imminent.
+  ///
+  /// This will become `true` as soon as the [dispose] method is called and will
+  /// become `false` once disposal actually starts (when [isDisposing] is
+  /// `true`).
+  bool get isDisposalImminent => _isDisposalImminent;
+
+  /// Whether the disposal of this object is imminent, in progress, or complete.
+  ///
+  /// This will become `true` as soon as the [dispose] method is called and will
+  /// remain `true` forever.
+  ///
+  /// This is equivalent to:
+  ///
+  ///     isDisposalImminent || isDisposing || isDisposed
+  ///
+  /// Recommended usage of this boolean is to guard public APIs such that all
+  /// calls after disposal has been requested (via [dispose]) are rejected:
+  ///
+  ///     Response sendRequest() async {
+  ///       if (isDisposalImminentOrComplete) {
+  ///         throw new StateError(
+  ///             'sendRequest() cannot be called, object is disposing');
+  ///       }
+  ///       ...
+  ///     }
+  bool get isDisposalImminentOrComplete =>
```

Comment:
Out nomenclature is getting complicated. Technically this should be `isDisposalImminentOrInProgressOrComplete` or something crazy. I'm fine with considering naming tech debt to be hashed out later.

Reply:
How bout `isOrWillBeDisposed`?

Reply:
Totally agree, I was struggling with that. Would it be worth scheduling a quick meeting to talk this through?